src/functions/searchResults/framework/database/query-builder.ts

```diff
@@ -1,18 +1,41 @@
 import * as mysql from 'mysql2';
+import { QueryParameters } from '../../domain/query_parameters';
 
-export enum DriverDetail {
-  ApplicationReference = 'application_reference',
-  DriverNumber = 'driver_number',
-}
+export const buildDriverDetailsSearchQuery = (queryParameters : QueryParameters): string => {
+  const parameterArray = [];
+  let queries = [];
 
-export const buildDriverDetailsSearchQuery = (
-  driverDetailKey: DriverDetail,
-  driverDetailValue: string,
-): string => {
-  const template = `
-  SELECT *
-  FROM TEST_RESULT
-  WHERE ? = ?;
-  `;
-  return mysql.format(template, [driverDetailKey, driverDetailValue]);
+  if (queryParameters.startDate && queryParameters.endDate) {
+    queries.push('test_date >= ? AND test_date <= ?');
+    parameterArray.push(queryParameters.startDate);
+    parameterArray.push(queryParameters.endDate);
+  }
+  if (queryParameters.driverId) {
+    queries.push('driver_number = ?');
+    parameterArray.push(queryParameters.driverId);
+  }
+  if (queryParameters.dtcCode) {
+    queries.push('json_extract(test_result, \'$.journalData.testCentre.costCode\') = ?');
+    parameterArray.push(queryParameters.dtcCode);
+  }
+  if (queryParameters.staffNumber) {
+    queries.push('staff_number = ?');
+    parameterArray.push(queryParameters.staffNumber);
+  }
+  if (queryParameters.appRef) {
+    queries.push('application_reference = ?');
+    parameterArray.push(queryParameters.appRef);
+  }
+
+  // Add AND between all statements
+  queries = [...queries].map((e, i) => i < queries.length - 1 ? [e, 'AND'] : [e]).reduce((a, b) => a.concat(b));
+
+  // Stringify the array, leaving spaces between
+  let queryString = 'SELECT * FROM TEST_RESULT WHERE ';
+  for (const query of queries) {
+    queryString += `${query} `;
```

Comment:
this should fail lint should concat the string together instead of doing a +=. You can also do queries.forEach(() => 'String Stuff Here')

Reply:
Don't think this actually falls foul of the `prefer-template` rule as it still technically uses a template however I would go for constructing it all in the template rather than using the `+=` operator. Would make it more readable :)